frontend/src/components/markers.ts

```diff
@@ -0,0 +1,103 @@
+import * as L from "leaflet";
+import { icon } from "@fortawesome/fontawesome-svg-core";
+import {
+  faBolt,
+  faBurst,
+  faCertificate,
+  faCircle,
+  faCloud,
+  faClover,
+  faDiamond,
+  faLocationArrow,
+  faLocationCrosshairs,
+  faLocationPin,
+  faMapPin,
+  faMound,
+  faSplotch,
+  faSquare,
+  faStar,
+  faStarOfLife,
+  faThumbTack,
+} from "@fortawesome/free-solid-svg-icons";
+
+/** choose shapes from font-awesome that look neutral (don't imply any good/bad
+ * connotation). put simpler shapes first. also be careful that similar looking
+ * shapes don't get assigned similar looking colors. */
+const icons = [
+  faCircle,
+  faSquare,
+  faDiamond,
+  faLocationPin,
+  faStarOfLife,
+  faSplotch,
+  faMound,
+  faBurst,
+  faClover,
+  faCertificate,
+  faStar,
+  faCloud,
+  faBolt,
+  faLocationArrow,
+  faMapPin,
+  faThumbTack,
+  faLocationCrosshairs,
+];
+
+/** https://tailwindcss.com/docs/customizing-colors */
+const colors = [
+  "#ef4444",
+  "#f97316",
+  "#f59e0b",
+  "#eab308",
+  "#84cc16",
+  "#22c55e",
+  "#10b981",
+  "#14b8a6",
+  "#06b6d4",
+  "#0ea5e9",
+  "#3b82f6",
+  "#6366f1",
+  "#8b5cf6",
+  "#a855f7",
+  "#d946ef",
+  "#ec4899",
+  "#f43f5e",
+].reverse();
+
+export const markerOptions = icons.map((def, index) => {
```

Comment:
Assembles a list of icons matched with colors, and converted into leaflet marker icons. Worth looking at this code a bit.

Reply:
Kudos for how thought-out the color cycling logic is; that's the kind of thing you wouldn't know expect through experience IMHO.